assignment.ts

```diff
@@ -0,0 +1,137 @@
+interface Admin {
+    name: string;
+    email: string;
+    createUser(user: User): void;
+  }
+  
+  interface Moderator {
+    name: string;
+    email: string;
+    banUser(user: User): void;
+  }
+  
+  interface RegularUser {
+    name: string;
+    email: string;
+    viewContent(): void;
+  }
```

Comment:
name and email are repeated here. You could hvae defined a base user interface and extended it in all of these to avoid the repetition.